Header header
string location
bool has_package